### PR TITLE
feat: adiciona turno 0 de declaração

### LINKED ISSUE #111: [Fase 3] Turno 0 — Declaração
## Parent

#99

## Contexto

No FDP, antes de jogar os turnos, cada jogador declara quantos turnos pretende fazer. Isso é o Turno 0. A declaração é sequencial anti-horária, cada jogador ouve os anteriores antes de declarar. Este slice introduz o fluxo de declaração no core e no adapter.

## O que construir

- Criar port `DecisorDeclaracao` em `src/core/portas/`
- Adicionar fases `aguardandoDeclaracao` e `processandoDeclaracao` ao `FasePartida`
- Na `Rodada`, após distribuir, entrar em `aguardandoDeclaracao`
- Fluxo: cada jogador declara em ordem anti-horária, armazenando `declaracoes: Record<string, number>` no estado
- Emitir evento `DECLARACAO_FEITA` por jogador
- Criar `DecisorDeclaracaoHumano` no adapter (input na UI)
- Criar `DecisorDeclaracaoBot` no adapter (heurística trivial — pode ser aleatório ou declarar 0)
- Após todos declararem, transicionar para `aguardandoJogada`

## Estratégia de testes

- **Vitest (TDD):** fluxo de declaração sequencial; declarações armazenadas no estado; evento emitido por jogador; ordem anti-horária respeitada; pode declarar 0
- **Validação visual:** antes de jogar, cada jogador declara (humano escolhe, bots declaram automaticamente), declarações visíveis na UI

## Fora do escopo

- Pontuação (|declarado − feito|) — slice #4
- Bots com temperatura (Fase 4)

## Critérios de aceite

- [ ] Port `DecisorDeclaracao` criado com interface `declarar(estado, mao): Promise<number>`
- [ ] Fluxo: após distribuir → `aguardandoDeclaracao` → cada jogador declara → `aguardandoJogada`
- [ ] Declarações sequenciais anti-horárias
- [ ] Campo `declaracoes` exposto no estado público
- [ ] Evento `DECLARACAO_FEITA` emitido por jogador
- [ ] `DecisorDeclaracaoHumano` pede input na UI
- [ ] `DecisorDeclaracaoBot` com heurística trivial
- [ ] Jogador pode declarar 0

### PR DESCRIPTION
**Pull request by Dhinihan**
## Resumo

- Cria o port `DecisorDeclaracao` e o fluxo de declaração sequencial na `Rodada`
- Expõe `declaracoes` no estado público e emite `DECLARACAO_FEITA`
- Adiciona decisores de declaração humano/bot e UI de seleção antes das jogadas
- Integra com as mudanças de empate vindas de `origin/main`

## Validação

- `pnpm lint`
- `pnpm typecheck`
- `pnpm test -- --run tests/core`
- `pnpm build`
- pre-push: `pnpm typecheck` e `pnpm test`

Closes #111

## Summary by CodeRabbit

* **New Features**
  * Added a declaration phase before card play with UI buttons and human confirm flow; AI opponents also declare.
* **Improvements**
  * Refined turn processing and winner resolution to handle asynchronous declarations and produce more consistent outcomes.
* **Tests**
  * Added comprehensive tests for declaration flow and card-comparison logic covering normal and edge cases.

### CODE REVIEW

File: src/adapters/phaser/scenes/JogoScene.ts [hunk 1]
```diff
@@ -48,33 +51,54 @@ export class JogoScene extends Scene {
...
+      decisorHumano: this.decisorDeclaracaoHumano,
+      atualizarIndicadorVez: this.atualizarIndicadorVez.bind(this),
+      iniciarTurnos: this.iniciarFluxoTurno.bind(this),
+    });
+  }
+  private async iniciarFluxoTurno(): Promise<void> {
+    await iniciarProcessamentoTurno({
+      cena: this,
+      rodada: this.rodada as Rodada,
+      labels: this.labels,
```
Comment (Dhinihan) [inline src/adapters/phaser/scenes/JogoScene.ts]: 🔴 **Bug crítico — labels stale causando crash**

Na linha 77, `labels: this.labels` captura a **referência ao array** no momento da chamada. Porém, em `redesenharTela()` (linha 125-134), `limparObjetos(this.objetos)` **destrói** os `Phaser.GameObjects.Text` do array via `.destroy()`, e depois `desenharMaos()` faz `this.labels = []` (linha 147) criando um **novo array** com labels vivos.

O `config.labels` capturado aqui continua apontando pro array velho com Text objects destruídos. Quando `atualizarLabelVencedor` (jogo-scene-loop.ts:82-88) roda após um turno com vencedor, ele chama `.setText()` num label morto → Phaser lança `cannot read properties of null (reading 'cut')` internamente.

**Por que não crasha em empate:** `atualizarLabelVencedor` tem `if (!vencedorId) return;` — no empate sai antes de tocar nos labels.

**No código antigo funcionava** porque o loop era método da scene e lia `this.labels` diretamente a cada iteração.

**Sugestão de fix:** passar getters em vez de valores:
```ts
getLabels: () => this.labels,
getDirecoesLabels: () => this.direcoesLabels,
```
Ou acessar `config.cena.labels` direto, já que `cena` é a `JogoScene`.

File: src/adapters/phaser/scenes/jogo-scene-loop.ts [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+}
+
+async function atualizarFimDeTurno(config: ConfigTurnos): Promise<void> {
+  const { rodada, turnoAnteriorRef } = config;
+  if (rodada.estado.turno <= turnoAnteriorRef.valor) return;
+  turnoAnteriorRef.valor = rodada.estado.turno;
+  const vencedorId = config.getVencedorTurno();
+  config.animarRecolhimento();
+  await esperar(config.cena, 800);
+  atualizarLabelVencedor({
```
Comment (Dhinihan) [inline src/adapters/phaser/scenes/jogo-scene-loop.ts:86]: ⚠️ **Este é o ponto exato do crash**

`config.labels` (linha 86) contém Text objects destruídos pelo `redesenharTela()`. A chamada `atualizarLabelVencedor(...)` tenta fazer `.setText()` num label morto → `null.setSize()` → exception.

A exception sobe, mas é engolida pelo `catch {}` vazio em `tentarJogarTurno` (linha 104-110), que retorna `false`, encerrando o loop silenciosamente. Resultado: jogo trava sem nenhuma mensagem de erro útil no console.

File: src/adapters/phaser/scenes/jogo-scene-loop.ts [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+export async function processarDeclaracoes(config: ConfigDeclaracoes): Promise<void> {
+  const { rodada } = config;
+  while (faseDeclaracao(rodada)) {
+    await prepararDeclaracaoAtual(config);
+    const declarou = await tentarDeclarar(rodada);
+    if (!declarou) return;
+    limparObjetosDeclaracao(config.objetos);
+    config.atualizarIndicadorVez();
+  }
+  if (rodada.estado.fase === 'aguardandoJogada') void config.iniciarTurnos();
```
Comment (Dhinihan) [inline src/adapters/phaser/scenes/jogo-scene-loop.ts]: ⚠️ **Fire-and-forget com `void` engole erros**

Se `config.iniciarTurnos()` lançar uma exception (como o bug de labels stale), o `void` descarta a Promise e o erro some silenciosamente. O callback `onCartaJogada` → `redesenharTela` roda entre turnos, destruindo labels que `iniciarTurnos` ainda referencia.

Sugestão: adicionar `.catch()` pelo menos pra logar:
```ts
void config.iniciarTurnos().catch((e) => console.error('Erro nos turnos:', e));
```

File: src/adapters/phaser/scenes/jogo-scene-loop.ts [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+async function tentarDeclarar(rodada: Rodada): Promise<boolean> {
+  try {
+    await rodada.declarar();
+    return true;
+  } catch {
+    return false;
+  }
+}
+
+async function tentarJogarTurno(rodada: Rodada): Promise<boolean> {
```
Comment (Dhinihan) [inline src/adapters/phaser/scenes/jogo-scene-loop.ts:109]: 💡 **`catch {}` vazio esconde bugs**

O catch vazio engole qualquer erro — incluindo o crash de labels stale. Durante desenvolvimento, isso dificulta muito o debugging. Sugestão:
```ts
} catch (e) {
  console.error('Falha em jogarTurno:', e);
  return false;
}
```
O mesmo se aplica a `tentarDeclarar` na linha 95.

File: src/adapters/phaser/scenes/JogoScene.ts [hunk 5]
```diff
@@ -28,18 +29,20 @@ const JOGADORES: Jogador[] = [
...
-  private mesaObjetos: Phaser.GameObjects.GameObject[] = [];
-  private destaque: EstadoDestaque = {};
-  private rodada?: Rodada;
-  private labels: Phaser.GameObjects.Text[] = [];
-  private direcoesLabels: ('horizontal' | 'vertical')[] = [];
-  private tweenVez?: Phaser.Tweens.Tween;
-  private turnoAnterior = 1;
-  private vencedorTurno?: string;
-  private manilhaObjetos: Phaser.GameObjects.GameObject[] = [];
+  objetos: Phaser.GameObjects.GameObject[] = [];
```
Comment (Dhinihan) [inline src/adapters/phaser/scenes/JogoScene.ts]: 💡 **Propriedades `private` → `public` sem necessidade**

Várias propriedades que eram `private` ficaram `public` (ex: `objetos`, `rodada`, `labels`, `destaque`). Isso expõe estado interno da scene pra qualquer código externo.

Se a mudança foi pra permitir acesso via `config.cena.labels` no loop, uma alternativa mais limpa seria passar getters/bound methods em vez de expor a propriedade:
```ts
getLabels: () => this.labels,
```
Assim as propriedades voltam a ser `private` e o loop não precisa acessar internals da scene.

**Review by Dhinihan — COMMENTED** [review]
## 🔍 Code Review — Hermes Agent

Encontrei **1 bug crítico** que causa o travamento que você reportou, além de **2 warnings** e **2 sugestões**. Detalhes nos inline comments.

### Resumo

🔴 **Bug crítico:** `config.labels` capturado por valor na linha 77 — fica stale após `redesenharTela()` destruir e recriar os labels. Causa o `cannot read properties of null (reading 'cut')` no turno 2+ quando há vencedor.

⚠️ **Warnings:** erros engolidos silenciosamente por `catch {}` vazio e `void` fire-and-forget.

💡 **Sugestões:** ref acoplada em `turnoAnteriorRef`, propriedades `private` → `public` sem necessidade.
